### ISSUE TITLE
Deprecation warning about a synchronous request

### ISSUE THREAD

**Issue by VSevagen** [chaoss/grimoirelab-sortinghat#531]
There is a warning about a synchronous request being made in `ui/src/main.js` line 17.
 According to the warning, it says, I quote,  `[Deprecation] Synchronous XMLHttpRequest on the main thread is deprecated because of its detrimental effects to the end user's experience.`
The solution would be simply to change our current request to `async.` However, since the request was made `sync` from the beginning, I would assume there is a purpose. In any case, I have my doubts on this so I created this issue.

**Comment by evamillan** [comment]
The request is explicitly synchronous because it needs to get the CSRF token from the Django server before the Apollo configuration is loaded. However, this could also possibly be achieved with an async fetch request, setting up the configuration within a `.then()` handler after the request has successfully retrieved the token.

**Comment by VSevagen** [comment]
Right. Tried it out actually and turns out that the warning is gone. 
So this is what I did:
- made a helper function `httpRequest` where we create out xhr request.
- Created a promise object and set the csrftoken declaration in `.then` which will only happen if request is resolved.
- Moved everything related to Apollo config to `.then` as well.

I refered to the following blog while doing it. https://gomakethings.com/promise-based-xhr/
